### PR TITLE
Feature/Refactor Fuel Information Related Vehicle Data

### PR DESCRIPTION
**Pull request by HSavynetska**
ATF Test Scripts to check #[3124](https://github.com/smartdevicelink/sdl_core/issues/3124)

This PR is **ready** for review.

### Summary
Scripts for "Refactor Fuel Information Related Vehicle Data" feature

### ATF version
develop

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: test_scripts/API/VehicleData/Refactor_Fuel_Information/006_OnVehicleData_with_only_one_param_in_fuelRange_structure.lua [hunk 1]
```diff
@@ -0,0 +1,38 @@
+---------------------------------------------------------------------------------------------------
```
Comment (GetmanetsIrina) [inline test_scripts/API/VehicleData/Refactor_Fuel_Information/006_OnVehicleData_with_only_one_param_in_fuelRange_structure.lua:1]: @GetmanetsIrina , please add checks of boundary and out of boundary values for GetVD and OnVD RCPs

**Comment by GetmanetsIrina** [comment]
@GetmanetsIrina , add checks with VEHICLEDATA_NOT_AVAILABLE resultCode for `fuelRange` data

**Comment by GetmanetsIrina** [comment]
@atiwari9, the update according to comments is performed in 4913b3bd4f4ba0cde41272f2109143e7e2d50811 and 6171755cf8e4a9698dfb44a3e5877e20e34c0aa8

**Comment by GetmanetsIrina** [comment]
Please add one more script with checking of IGNORED resultCode in case app is already subscribed (also for UnsubscribeVD)

**Comment by atiwari9** [comment]
Test Report for commit: [6171755](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2389/commits/6171755cf8e4a9698dfb44a3e5877e20e34c0aa8)

Logs: 
[Refactor_Fuel_Range_TestReport.zip](https://github.com/smartdevicelink/sdl_atf_test_scripts/files/4772686/Refactor_Fuel_Range_TestReport.zip)

```
TOTAL: 14
PASSED: 11 
FAILED: 0 
ABORTED: 0 
SKIPPED: 3 
07 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/007_VD_parameter_in_API_DB_app_version_less_than_parameter_version.lua https://github.com/smartdevicelink/sdl_core/issues/3309
13 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/013_SubscribeVD_not_success_codes_for_fuelRange_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
14 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/014_UnsubscribeVD_not_success_codes_for_fuelRange_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
```

**Comment by GetmanetsIrina** [comment]
Scripts with `IGNORED` resultCode are added in aa47ade

**Comment by atiwari9** [comment]
Final test report for commit: [aa47ade](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2389/commits/aa47ade5d5af8b720df0212f8c4b370c1f9fb049)
```
=====================================================================================================
TOTAL: 16
PASSED: 13 
FAILED: 0 
ABORTED: 0 
SKIPPED: 3 
07 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/007_VD_parameter_in_API_DB_app_version_less_than_parameter_version.lua https://github.com/smartdevicelink/sdl_core/issues/3309
13 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/013_SubscribeVD_not_success_codes_for_fuelRange_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
14 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/014_UnsubscribeVD_not_success_codes_for_fuelRange_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
=====================================================================================================
```

**Comment by LitvinenkoIra** [comment]
@atiwari9 please review this PR

**Comment by atiwari9** [comment]
Retest Report for atf scripts [aa47ade](https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2389/commits/aa47ade5d5af8b720df0212f8c4b370c1f9fb049):

```
=====================================================================================================
TOTAL: 16
PASSED: 13 
FAILED: 0 
ABORTED: 0 
SKIPPED: 3 
07 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/007_VD_parameter_in_API_DB_app_version_less_than_parameter_version.lua https://github.com/smartdevicelink/sdl_core/issues/3309
13 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/013_SubscribeVD_not_success_codes_for_fuelRange_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
14 ;./test_scripts/API/VehicleData/Refactor_Fuel_Information/014_UnsubscribeVD_not_success_codes_for_fuelRange_data.lua https://github.com/smartdevicelink/sdl_core/issues/3384
=====================================================================================================
```

Logs:
[Retest_Report_Fuel_Range_aa47ade.zip](https://github.com/smartdevicelink/sdl_atf_test_scripts/files/4800519/Retest_Report_Fuel_Range_aa47ade.zip)

**Comment by atiwari9** [comment]
PR Approved.

**Comment by LitvinenkoIra** [comment]
@theresalech PR is ready for Livio review. Thank you!

File: test_scripts/API/VehicleData/Refactor_Fuel_Information/common.lua [hunk 2]
```diff
@@ -0,0 +1,159 @@
+---------------------------------------------------------------------------------------------------
```
Comment (ShobhitAd) [inline test_scripts/API/VehicleData/Refactor_Fuel_Information/common.lua:1]: I don't think that this vehicle data item should have its own separate common file and it should instead use commonVehicleData.lua

File: test_scripts/API/VehicleData/Refactor_Fuel_Information/001_GetVehicleData_fuelRange_all_params.lua [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+--[[ Scenario ]]
+common.Title("Preconditions")
+common.Step("Clean environment", common.preconditions)
+common.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+common.Step("Register App", common.registerApp)
+common.Step("PTU", common.policyTableUpdate, { common.pTUpdateFunc })
+common.Step("Activate App", common.activateApp)
+
+common.Title("Test")
+common.Step("App sends GetVehicleData for fuelRange", common.getVehicleData, { { common.allVehicleData } })
```
Comment (ShobhitAd) [inline test_scripts/API/VehicleData/Refactor_Fuel_Information/001_GetVehicleData_fuelRange_all_params.lua:24]:
```suggestion
common.Step("GetVehicleData for fuelRange", common.getVehicleData, { { common.allVehicleData } })
```

File: test_scripts/API/VehicleData/Refactor_Fuel_Information/002_GetVehicleData_with_one_param_with_boundary_value_in_fuelRange_strucrure.lua [hunk 4]
```diff
@@ -0,0 +1,31 @@
+---------------------------------------------------------------------------------------------------
```
Comment (ShobhitAd) [inline test_scripts/API/VehicleData/Refactor_Fuel_Information/002_GetVehicleData_with_one_param_with_boundary_value_in_fuelRange_strucrure.lua:1]: File should be renamed to `002_GetVehicleData_with_one_param_with_boundary_value_in_fuelRange_structure.lua`

File: test_scripts/API/VehicleData/Refactor_Fuel_Information/004_OnVehicleData_with_new_fuelRange_structure.lua [hunk 5]
```diff
@@ -0,0 +1,30 @@
+---------------------------------------------------------------------------------------------------
```
Comment (ShobhitAd) [inline test_scripts/API/VehicleData/Refactor_Fuel_Information/004_OnVehicleData_with_new_fuelRange_structure.lua:1]: Referring to this comment https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#discussion_r451132043, I think this parameter should be covered in the `OnVehicleData/001_Success_flow.lua` test and this new test should be removed

File: test_scripts/API/VehicleData/Refactor_Fuel_Information/007_VD_parameter_in_API_DB_app_version_less_than_parameter_version.lua [hunk 6]
```diff
@@ -0,0 +1,78 @@
+---------------------------------------------------------------------------------------------------
```
Comment (ShobhitAd) [inline test_scripts/API/VehicleData/Refactor_Fuel_Information/007_VD_parameter_in_API_DB_app_version_less_than_parameter_version.lua:1]: Referring to this comment https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#discussion_r451132043, I think this new test should be removed and the existing test `VehicleDataVersioning/003_VD_parameter_in_API_DB_app_version_less_then_parameter_version.ua` should be expanded to test for the `FuelRange` params

File: test_scripts/API/VehicleData/Refactor_Fuel_Information/009_App_version_is_equal_to_fuelRange_parameter_version_is_listed_in_DB_and_API.lua [hunk 7]
```diff
@@ -0,0 +1,39 @@
+---------------------------------------------------------------------------------------------------
```
Comment (ShobhitAd) [inline test_scripts/API/VehicleData/Refactor_Fuel_Information/009_App_version_is_equal_to_fuelRange_parameter_version_is_listed_in_DB_and_API.lua:1]: Referring to this comment https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#discussion_r451132043, I think this new test should be removed and the existing test `VehicleDataVersioning/002_VD_parameter_in_API_DB_app_version_equal_parameter_version.lua` should be expanded to include the fuelRange param

**Comment by LitvinenkoIra** [comment]
@ShobhitAd Thank you for the review, we will prepare the update according to this https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#discussion_r451132043 in all Vehicle Data related PR's including this one. This activity is planned for the next sprint and we will notify you when the PR will be ready for re-review

**Comment by ShobhitAd** [comment]
@LitvinenkoIra just as a note, in reference to https://github.com/smartdevicelink/sdl_atf_test_scripts/pull/2413#issuecomment-664457788, the updates to this PR should be based off of the `feature/vehicle_data_testing` branch.

**Comment by dboltovskyi** [comment]
@ShobhitAd Please notice as was agreed previously current PR has been completely reworked based on `feature/vehicle_data_testing` branch and is ready for PM review.

**Pull request merged**